tests/integration/test_cli.py

```diff
@@ -99,3 +118,90 @@ def test_consume_halt_with_exitcode(callback, exit_code, msg, queue, cli_conf):
 
     assert process.returncode == exit_code, process.stderr.read()
     assert msg in process.stdout.read()
+
+
+@pytest_twisted.inlineCallbacks
+def test_consume_monitoring(queue, cli_conf, available_port):
+    """Assert the monitoring works."""
+    cmd = shutil.which("fedora-messaging")
+    args = [
+        sys.executable,
+        cmd,
+        f"--conf={cli_conf}",
+        "consume",
+        "--callback=tests.integration.test_cli:execute_action",
+        f"--queue-name={queue}",
+        "--exchange=amq.topic",
+        "--routing-key=#",
+    ]
+
+    http_client = Agent(reactor)
+    base_url = f"http://localhost:{available_port}".encode("ascii")
+
+    # Monitoring not available yet
+    with pytest.raises(error.ConnectionRefusedError):
+        yield http_client.request(b"GET", base_url + b"/live")
+
+    # Start the consumer
+    process = subprocess.Popen(args)  # noqa: S603
+
+    # Wait for the consumer to start up
+    for _ in range(5):
+        yield sleep(1)
+        try:
+            response = yield http_client.request(b"GET", base_url + b"/live")
+        except error.ConnectionRefusedError:
+            continue
+        # Check the monitoring on startup
+        body = yield readBody(response)
+        assert body == b'{"status": "OK"}\n'
+        response = yield http_client.request(b"GET", base_url + b"/ready")
+        body = yield readBody(response)
+        assert body == (
+            b'{"consuming": true, "published": 0, "consumed": {"received": 0, "processed": 0, "dropped": 0, '
+            b'"rejected": 0, "failed": 0}}\n'
+        )
+        break
+    else:
+        pytest.fail(f"Monitoring didn't start: {process.stdout.read()} -- {process.stderr.read()}")
+
+    # Publish a message
+    yield threads.deferToThread(api.publish, message.Message())
+    yield sleep(0.5)
+
+    # Check stats
+    response = yield http_client.request(b"GET", base_url + b"/ready")
+    body = yield readBody(response)
+    assert body == (
+        b'{"consuming": true, "published": 0, "consumed": {"received": 1, "processed": 1, "dropped": 0, '
+        b'"rejected": 0, "failed": 0}}\n'
+    )
+
+    # Publish a message and drop it
+    yield threads.deferToThread(api.publish, message.Message(body={"action": "drop"}))
+    yield sleep(0.5)
+
+    # Check stats
+    response = yield http_client.request(b"GET", base_url + b"/ready")
+    body = yield readBody(response)
+    assert body == (
+        b'{"consuming": true, "published": 0, "consumed": {"received": 2, "processed": 1, "dropped": 1, '
+        b'"rejected": 0, "failed": 0}}\n'
+    )
+
+    # Don't check the reject action or the message will be put back in the queue and loop forever
+    # Don't check a generic processing failure because it stops the consumer instantly and we can't look
```

Comment:
I guess it could log the final stats on exit, if available, and then you could inspect the log, but whether you bother with all that is totally up to you.